javascript/ql/src/Security/CWE-327/BadRandomness.ql

```diff
@@ -99,6 +99,7 @@ DataFlow::Node badCrypto(string description) {
     description = "modulo" and
     goodRandom() = random and
     random.flowsToExpr(mod.getLeftOperand()) and
+    // division by a power of 2 is OK. E.g. if `x` is uniformly random is in the range [0..255] then `x % 32` is uniformly random in the range [0..31]. 
```

Comment:
Let's have a look a the false positives for this before we do more, but I am a bit worried about the implicit assumption that the input range is a power of two already.

But, on the other hand, if a library do support outputs that are uniform in a specific range such as `[0,99]` or `[0, 63]`, then the programmer probably wouldn't need to do this modulo manually.

Reply:
I'll remove the "power of 2" constraint for the full LGTM run, but keep it in the query.